File: HouseFinder_API/BusinessObjects/Report.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace BusinessObjects
{
    public partial class Report
    {
        public int ReportId { get; set; }
        public string ReportContent { get; set; }
        public string StudentId { get; set; }
        public int HouseId { get; set; }
        public int StatusId { get; set; }
        public bool Deleted { get; set; }
        public DateTime ReportedDate { get; set; }
        public DateTime? SolvedDate { get; set; }
        public string SolvedBy { get; set; }

        public virtual House House { get; set; }
        public virtual User SolvedByNavigation { get; set; }
        public virtual ReportStatus Status { get; set; }
        public virtual User Student { get; set; }
    }
}

File: house-finder-client/src/app/models/report.ts
import { Staff } from "./staff";

//DTO: Report
export interface Report {
    reportContent: string;
    studentId: string;
    houseId: number;
    statusId: number;
    deleted: boolean;
    reportedDate: Date;
    solvedDate?: Date;
    solvedByNavigation?: Staff;
    //no need for createdDate & lastModifiedDate
}
